tests/watch.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { proxy } from '../src/vanilla'
+import { watch } from '../src/utils'
+
+describe('watch', () => {
+  it('should re-run for individual proxy updates', async () => {
+    const reference = proxy({ value: 'Example' })
+
+    const callback = jest.fn()
+
+    watch((get) => {
+      get(reference)
+      callback()
+    })
+
+    expect(callback).toBeCalledTimes(1)
+    reference.value = 'Update'
+    await Promise.resolve()
+    expect(callback).toBeCalledTimes(2)
+  })
+  it('should re-run for multiple proxy updates', async () => {
+    const A = proxy({ value: 'A' })
+    const B = proxy({ value: 'B' })
+
+    const callback = jest.fn()
+
+    watch((get) => {
+      get(A)
+      get(B)
+      callback()
+    })
+
+    expect(callback).toBeCalledTimes(1)
+    A.value = 'B'
+    await Promise.resolve()
+    expect(callback).toBeCalledTimes(2)
+    B.value = 'C'
+    await Promise.resolve()
+    expect(callback).toBeCalledTimes(3)
+  })
+  it('should cleanup when state updates', async () => {
+    const reference = proxy({ value: 'Example' })
+
+    const callback = jest.fn()
+
+    watch((get) => {
+      get(reference)
+
+      return () => {
+        callback()
+      }
+    })
+
+    expect(callback).toBeCalledTimes(0)
+    reference.value = 'Update'
+    await Promise.resolve()
+    expect(callback).toBeCalledTimes(1)
+  })
+  it('should cleanup when stopped', () => {
+    const callback = jest.fn()
+
+    const stop = watch(() => callback)
+
+    expect(callback).toBeCalledTimes(0)
+    stop()
+    expect(callback).toBeCalledTimes(1)
+  })
+  it('should cleanup recursively when stopped', () => {
+    const callback = jest.fn()
+
+    const stop = watch(() =>
+      watch(() => () => {
+        callback()
+      })
+    )
```

Comment:
```suggestion
    const stop = watch(() =>
      watch(() => () => {
        callback()
      })
      return () => {
        outerCallback()
      }
    )
```

Would it be nice to add this to make sure it doesn't override, but merge cleanups?

What would be the real use case? I think you have something in your mind. It would be nice to have that in tests.

Reply:
I could probably replace it with a `subscribe` example instead of just `watch` just to demonstrate a cleanup.